cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/service/checker/AbstractGcpBaseResourceChecker.java

```diff
@@ -30,25 +30,53 @@ protected CloudResource createNamedResource(ResourceType type, String name) {
     }
 
     protected void exceptionHandler(GoogleJsonResponseException ex, String name, ResourceType resourceType) {
-        throwIfNull(ex, () -> new IllegalArgumentException("Unable to handle exception due to: " + GoogleJsonResponseException.class.getSimpleName()
-                + " should not be null!"));
-        GoogleJsonError jsonError = ex.getDetails();
-        LOGGER.debug("{} contains the following {}: {}",
-                GoogleJsonResponseException.class.getSimpleName(),  GoogleJsonError.class.getSimpleName(), jsonError != null ? jsonError.toString() : "null");
+        GoogleJsonError jsonError = getGoogleJsonError(ex);
+        if (jsonError != null) {
+            if (jsonError.get(HTTP_CODE_KEY) != null && jsonError.get(HTTP_CODE_KEY).equals(HttpStatus.SC_NOT_FOUND)) {
+                LOGGER.info("Resource {} not found: {}", resourceType, name);
+            } else {
+                throw extractedPermissionIssue(ex, jsonError);
+            }
+        } else {
+            throw extractIfNoJson(ex);
+        }
+    }
+
+    protected GcpResourceException exceptionHandlerWithThrow(GoogleJsonResponseException ex, String name, ResourceType resourceType) {
```

Comment:
name is misleading as this method does not throw, as opposed to the one above, which throws
it should be named that suggest this method produces a new exception instance